R/hp_spacyr.R

```diff
@@ -1,28 +1,80 @@
-#' package. More specifically, we rely on `spacyr::spacy_parse()` to tokenize and
-#' tag the string provided. Note that this functionality requires the `spacyr`
-#' package.
+#' Creating a hierplane object from text is made possible by employing
+#' the `spacyr` package. More specifically, the `spacyr::spacy_parse()` function
+#' is used to tokenize and tag the sentence(s) provided. Note that this
+#' functionality requires the `spacyr` package.
 #'
-#' @param .data A string of text.
-#' @param settings Assign your dataframes columns to hierplane variables.
+#' If more than one sentence is detected in the input text, a list of hierplane
+#' objects is returned.
+#'
+#' @param .data A sentence.
+#' @param node_type `spacyr::spacy_parse()` variable to be used as node type (used for plane styling).
+#' @param link `spacyr::spacy_parse()` variable to be used to generate lane tags (i.e. connections between planes).
+#' @param attributes `spacyr::spacy_parse()` variable(s) to be used for generating the attribute tags.
+#' See full list of available attributes here: [](https://spacy.io/api/token#attributes)
 #' @param styles Assign styles to hierplane.
-#' @param ... Additional parameters to pass to `spacyr::spacy_parse`
+#' @param ... Additional parameters to pass to `spacyr::spacy_parse()`
+#'
+#' @examples
+#' if (FALSE) {
+#'
+#'   # Custom attribute tags
+#'   hp_spacyr("I was in college 25 years ago.", attributes = c("pos", "is_stop")) %>%
+#'     hierplane()
+#'
+#'   # Multiple sentences in input
+#'   hp_spacyr("I have a cat. Her name is mocha. She has a round belly.") %>%
+#'     lapply(hierplane)
+#'
+#' }
+#'
 #' @md
 #' @export
-hp_spacyr <- function(.data, settings = spacyr_default(), styles = NULL, ...) {
+hp_spacyr <- function(.data,
+                      node_type = "dep_",
+                      link = "dep_",
+                      attributes = spacyr_default_attributes(),
+                      styles = NULL, ...) {
+
   requireNamespaceQuietStop("spacyr")
-  x <- build_tree(
-    x = transform_logical(spacyr_df(.data, ...)),
-    title = .data,
-    settings = settings,
-    styles = styles
+
+  .data <- gsub("\\s+", " ", .data)
+
+  settings <- hierplane_settings(
+      type = "spacyr",
+      parent_id = "head_token_id",
+      child_id = "token_id",
+      child = "token",
+      node_type = node_type,
+      link = link,
+      root_tag = "ROOT",
+      attributes = attributes
+  )
+
+  additional_attributes <- setdiff(
+    c(node_type, link, attributes),
+    spacyr_default_cols()
   )
-  structure(x, class = c("hierplane_tree", "hierplane_spacyr", class(x)))
+
+  sents <- get_sents(.data)
+
+  out <- lapply(sents, function(sent) {
+    x <- build_tree(
+      x = transform_logical(spacyr_df(sent, additional_attributes, ...)),
+      title = sent,
+      settings = settings,
+      styles = styles
+    )
+
+    structure(x, class = c("hierplane_tree", "hierplane_spacyr", class(x)))
```

Comment:
I wonder multi sentence hierplanes should name the list for clarity for example:
```r
# maybe this
$sentence1
<hierplane_tree object>
$sentence2
<hierplane_tree object>
$sentence3
<hierplane_tree object>

# instead of this?
[[1]]
<hierplane_tree object>
[[2]]
<hierplane_tree object>
[[3]]
<hierplane_tree object>
```

Reply:
Great idea! I'll go ahead and add.